pyrealm/pmodel/new_subdaily.py

```diff
@@ -0,0 +1,285 @@
+    :math:`V_{cmax25}` and :math:`J_{max25}`.
+
+    The first dimension of the data arrays use to create the
+    :class:`~pyrealm.pmodel.pmodel_environment.PModelEnvironment` instance must
+    represent
+    the time
+    axis of the observations. The actual datetimes of those observations must then be
+    used to initialiase a :class:`~pyrealm.pmodel.fast_slow_scaler.FastSlowScaler`
+    instance, and one of the ``set_`` methods of that class must be used to define an
+    acclimation window.
+
+    The workflow of the model is then:
+
+    * The daily acclimation window set in the
+      :class:`~pyrealm.pmodel.fast_slow_scaler.FastSlowScaler` instance is used to
+      calculate daily average values for forcing variables from within the acclimation
+      window.
+    * A standard P Model is then run on those daily forcing values to generate predicted
+      states for photosynthetic parameters that give rise to optimal productivity.
+    * The :meth:`~pyrealm.pmodel.subdaily.memory_effect` function is then used to
+      calculate realised slowly responding values for :math:`\xi`, :math:`V_{cmax25}`
+      and :math:`J_{max25}`, given a weight :math:`\alpha \in [0,1]` that sets the speed
+      of acclimation. The ``handle_nan`` argument is passed to this function to set
+      whether missing values in the optimal predictions are permitted and handled.
+    * The realised values are then filled back onto the original subdaily timescale,
+      with :math:`V_{cmax}` and :math:`J_{max}` then being calculated from the slowly
+      responding :math:`V_{cmax25}` and :math:`J_{max25}` and the actual subdaily
+      temperature observations and :math:`c_i` calculated using realised values of
+      :math:`\xi` but subdaily values in the other parameters.
+    * Predictions of GPP are then made as in the standard P Model.
+
+    Args:
+        env: An instance of
+          :class:`~pyrealm.pmodel.pmodel_environment.PModelEnvironment`
+        fs_scaler: An instance of
+          :class:`~pyrealm.pmodel.fast_slow_scaler.FastSlowScaler`.
+        fapar: The :math:`f_{APAR}` for each observation.
+        ppfd: The PPDF for each observation.
+        alpha: The :math:`\alpha` weight.
+        handle_nan: Should the :func:`~pyrealm.pmodel.subdaily.memory_effect` function
+          be allowed to handle missing values.
+        kphio: The quantum yield efficiency of photosynthesis (:math:`\phi_0`, -).
+        fill_kind: The approach used to fill daily realised values to the subdaily
+          timescale, currently one of 'previous' or 'linear'.
+    """
+
+    def __init__(
+        self,
+        env: PModelEnvironment,
+        fs_scaler: FastSlowScaler,
+        fapar: NDArray,
+        ppfd: NDArray,
+        kphio: float = 1 / 8,
+        do_ftemp_kphio: bool = True,
+        method_optchi: str = "prentice14",
+        method_jmaxlim: str = "wang17",
+        alpha: float = 1 / 15,
```

Comment:
Would it make sense to mention the assumptions regarding keeping the alpha value = 1/15 ?

Reply:
I've expanded the explanation in the docstrings.